### DISCUSSION TITLE
FlareSolverr integration (bypass Cloudflare protected sites)

### DISCUSSION THREAD

**Discussion by adan89lion** [go-shiori/shiori#828] [Feature requests]
[FlareSolverr](https://github.com/FlareSolverr/FlareSolverr) provides a friendly backend to bypass websites with Cloudflare captcha protection. As more and more articles I save are blocked by Cloudflare, I think a FlareSolverr can be important and essential part of Shiori.

Also possibly solve #454.

**Comment by Monirzadeh** [comment]
Do you test to pass shiori trafic through FlareSolverr with environment variable?
As I work with FlareSolverr is not a stable solution to bypass Cloudflare.  do you know any better alternative?

If you want FlareSolverr for specific domain You can combine FlareSolverr with [Privoxy](https://www.privoxy.org/) too.
```mermaid
flowchart LR
Shiori --> Privoxy --> Flaresolverr --> Internet
Privoxy -->|Domain you don't want pass to Flaresolverr | Internet
```

**Comment by fmartingr** [comment]
While I'm not completely against helping reaching captcha protected pages, it can get very time consuming since those evolve really quickly. I prefer some setup like @Monirzadeh mentioned where you have a proxy in front of Shori (or any service you need it for) to handle that kind of scenarios.

We could try and document it as a nice-to-have feature, but not officially supported.

**Comment by Oppen** [comment]
On Wallabag they follow a different approach that:
1. Doesn't break with time.
2. Works always.
3. Should be easy to implement (TM).
They simply have an option for the bookmark request to carry the contents as payload, so instead of Shiori making a request to the bookmarked site your browser provides the rendered contents. This is of course more bandwidth intensive, so it's off by default.

**Reply by Monirzadeh** [reply]
another things i thinking about is a way to pass paywall
for example Medium article can be behind paywall but some service can provide article of them.
but i am not sure what is the best way to add them to that . i mean shiori should can use this type of service but not struggle to handle something like paywall itself.
